lib/traject/marc_extractor.rb

```diff
@@ -264,17 +267,17 @@ def extract(marc_record)
     #
     # Third (optional) arg to block is self, the MarcExtractor object, useful for custom
     # implementations.
-    def each_matching_line(marc_record)
-      marc_record.fields(@interesting_tags_hash.keys).each do |field|
+    def each_matching_line(marc_record,tags=nil)
+      tags ||= @interesting_tags
+      marc_record.each_by_tag(tags) do |field|
```

Comment:
each_matching_line is meant to be public API that you can call on a MarcExtractor to do custom things. I don't understand what this second arg `tags` is, it hasn't been documented, and I think it ruins it as API -- it's probably something the MarcExtractor already knows about, so doesn't make that much sense as API? It still works the same if called without this optional arg? That might be okay then. But it needs to be documented in comments what it is, and why it's been made an arg, and why you'd ever choose to pass in a second arg instead of leaving it the default.  There are a few other places in this PR